src/linter.cc

```diff
@@ -57,140 +59,162 @@ absl::Status PrintASTTree(absl::string_view sql) {
+    byte_position = location.byte_position();
+    absl::Status status = ParseNextScriptStatement(&location, ParserOptions(),
+                                                   &output, &is_the_end);
+    if (!status.ok()) {
+      // TODO(orhanuysal): Implement a token parser to seperate statements
+      // currently, when parser fails, it is unable to determine
+      // the end of the statement.
+      result.Add(ErrorCode::kParseFailed, sql, byte_position, "Parser Failed");
+      break;
+    }
   }
-  return absl::OkStatus();
+  return result;
 }
 
-absl::Status CheckSemicolon(absl::string_view sql) {
+LinterResult CheckSemicolon(absl::string_view sql) {
+  LinterResult result;
   std::unique_ptr<ParserOutput> output;
 
   ParseResumeLocation location = ParseResumeLocation::FromStringView(sql);
   bool is_the_end = false;
 
   while (!is_the_end) {
-    ZETASQL_RETURN_IF_ERROR(ParseNextScriptStatement(&location, ParserOptions(),
-                                                     &output, &is_the_end));
-
+    absl::Status status = ParseNextScriptStatement(&location, ParserOptions(),
+                                                   &output, &is_the_end);
+    if (!status.ok()) return LinterResult(status);
     int location =
         output->statement()->GetParseLocationRange().end().GetByteOffset();
 
     if (location >= sql.size() || sql[location] != ';') {
-      return absl::Status(absl::StatusCode::kFailedPrecondition,
-                          "Each statemnt should end with a consequtive"
-                          "semicolon ';'");
+      result.Add(ErrorCode::kSemicolon, sql, location,
+                 "Each statemnt should end with a consequtive"
+                 "semicolon ';'");
     }
   }
-  return absl::OkStatus();
+  return result;
 }
 
-bool AllUpperCase(const absl::string_view &sql,
-                  const ParseLocationRange &range) {
+bool ConsistentUppercaseLowercase(const absl::string_view &sql,
+                                  const ParseLocationRange &range) {
+  bool uppercase = false;
+  bool lowercase = false;
   for (int i = range.start().GetByteOffset(); i < range.end().GetByteOffset();
        ++i) {
-    if ('a' <= sql[i] && sql[i] <= 'z') return false;
+    if ('a' <= sql[i] && sql[i] <= 'z') lowercase = true;
+    if ('A' <= sql[i] && sql[i] <= 'Z') uppercase = true;
   }
-  return true;
+  // There shouldn't be any case any Keyword
+  // contains both uppercase and lowercase characters
+  return !(lowercase && uppercase);
 }
 
-absl::Status CheckUppercaseKeywords(absl::string_view sql) {
+LinterResult CheckUppercaseKeywords(absl::string_view sql) {
   ParseResumeLocation location = ParseResumeLocation::FromStringView(sql);
   std::vector<ParseToken> parse_tokens;
-  absl::Status tokenizer_status =
+  LinterResult result;
+
+  absl::Status status =
       GetParseTokens(ParseTokenOptions(), &location, &parse_tokens);
 
+  if (!status.ok()) return LinterResult(status);
```

Comment:
Same with parsing error: We actually know a broken token location.

Reply:
It will also be solved with proper status printing.